flink-connectors/flink-orc/src/main/java/org/apache/flink/orc/OrcColumnarRowInputFormat.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.orc;
+
+import org.apache.flink.core.fs.FileInputSplit;
+import org.apache.flink.orc.vector.AbstractOrcColumnVector;
+import org.apache.flink.table.dataformat.ColumnarRow;
+import org.apache.flink.table.dataformat.vector.VectorizedColumnBatch;
+
+import org.apache.hadoop.conf.Configuration;
+import org.apache.orc.TypeDescription;
+
+import java.io.IOException;
+import java.util.Arrays;
+
+/**
+ * InputFormat to read ORC files into {@link ColumnarRow}.
+ */
+public class OrcColumnarRowInputFormat extends OrcInputFormat<ColumnarRow> {
```

Comment:
Shall you add a usage scenario for this `OrcColumnarRowInputFormat`?
Like a table source based on ` OrcColumnarRowInputFormat`.

Reply:
No, I don't want to add orc table source, after FileSystem support, I will integrate orc to it.

Reply:
So there will be one filesystem for all(avro, csv, orc, parquet, etc) and no more individual table sources any more？

Reply:
Yes.